Minor optimizations and cleanup

seanbright opened the pull request.

bsutton commented:
Sean firstly thanks for the patch.
Couple of questions.
I'm concerned about the changes from concurrentmaps.
Particularly the activeBridges.

The comment states:
// this map is shared (static) so that connections to asterisk that attach
    // late can
    // benefit from the known bridge information of older connections
    final static Map<String, BridgeState> activeBridges = new ConcurrentHashMap<>();

The member is not private so I'm not certain how you can guarantee that it won't be access by another thread?

seanbright commented:
@bsutton Good catch. I glossed over the fact that it was a static member variable (although I'm a little skeptical of the use-case). I've switched it back to a CHM and updated a CAS in BridgesActive.createBridge.

I left the Map in BridgeState as a HashMap and am doing good ol' synchronized blocks because there are a few places where we are doing a check-then-act that you wouldn't get for free with a CHM and you'd have to externally synchronize anyway.
